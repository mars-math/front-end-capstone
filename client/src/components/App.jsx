import React from 'react';
import QandA from './QandA/QandA.jsx';
import AllReviews from './RatingsReview/AllReviews.jsx';
import ProductInfo from './Overview/SM-ProductInfo.jsx';
import CardCarousel from './RelatedItemsAndOutfit/CardCarousel.jsx';
import RelatedItems from './RelatedItemsAndOutfit/RelatedItems.jsx';
import OutfitList from './RelatedItemsAndOutfit/OutfitList.jsx';
import Overview from './Overview/SM-Overview.jsx';
import RatingBreakdown from './RatingsReview/RatingBreakdown.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      url: window.location.pathname.slice(8).replaceAll('/', ''),
    };
  }

  render() {
    return (
      <>
        <h1>Kloth</h1>
        <Overview url={this.state.url} />
        <RelatedItems overviewId={this.state.url} />
        <OutfitList overviewId={this.state.url} />
        <AllReviews url={this.state.url} />
        <div>
          <QandA url={this.state.url} />
        </div>
      </>
    );
  }
}

export default App;

/*
1. Overview
2. Reviews
3. Q&A
4. Related/Outfit
*/
