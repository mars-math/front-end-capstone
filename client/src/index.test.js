import React from 'react';
import ReactDOM from 'react-dom';
import App from './components/App.jsx';
import Price from './components/Overview/SM-Price.jsx';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';

//dummy test
// it('one is one', () => {
//   expect(1).toEqual(1)
// });


xit('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

xtest('it renders given category name', () => {
  render(<Price price='3'/>)
  expect(screen.getByText('$3')).toBeInTheDocument()
});
