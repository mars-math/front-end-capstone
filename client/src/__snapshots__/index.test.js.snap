// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    style={
      Object {
        "marginLeft": "13%",
      }
    }
  >
    KLOTH
  </h1>,
  <div>
    loading
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "marginLeft": "15%",
      }
    }
  >
    <h2>
      Related Items
    </h2>
  </div>,
  <div
    className="react-multi-carousel-list react-multi-carousel-max-width "
  >
    <ul
      className="react-multi-carousel-track react-multi-carousel-track-modified"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "marginLeft": "15%",
      }
    }
  >
    <h2>
      My Outfit
    </h2>
    <button
      onClick={[Function]}
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": "20px",
        }
      }
      type="button"
    >
      Add to Outfit
    </button>
  </div>,
  <div
    className="react-multi-carousel-list react-multi-carousel-max-width "
  >
    <ul
      className="react-multi-carousel-track react-multi-carousel-track-modified"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "fontSize": "20px",
        "justifyContent": "center",
        "marginLeft": "3%",
      }
    }
  >
    <div>
      0 reviews, sorted by
    </div>
    <select
      className="review-dropdown"
      onChange={[Function]}
    >
      <option
        value="most recent ▼"
      >
        most recent
      </option>
      <option
        value="most helpful ▼"
      >
        most helpful
      </option>
    </select>
  </div>,
  <div
    className="all-review-components"
    data-testid="allRev-1"
  >
    <div
      className="individual-reviews"
      style={
        Object {
          "maxHeight": "601px",
          "overflow": "auto",
          "width": "45%",
        }
      }
    >
      <button
        className="cust-button "
        onClick={[Function]}
        type="button"
      >
        Write New Review
      </button>
    </div>
    <div
      className="rating-breakdown"
    >
      <div
        className="total-stars-render"
        id="reviewsJump"
      >
        <div
          style={
            Object {
              "fontSize": "3em",
            }
          }
        >
          NaN
        </div>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 100 100"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50 0L63.1958 31.8376L97.5528 34.5491L71.3512 56.9374L79.3893 90.4509L50 72.45L20.6107 90.4509L28.6488 56.9374L2.44717 34.5491L36.8042 31.8376L50 0Z"
              fill="#C4C4C4"
            />
          </svg>
        </span>
      </div>
      <div
        style={
          Object {
            "fontSize": "1.4em",
            "fontWeight": "bold",
            "marginBottom": "10px",
            "marginTop": "10px",
          }
        }
      >
        0
         
        reviews
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="component-break"
  >
    <div
      className="main"
    >
      <h3>
        Questions & Answers
      </h3>
      <div
        data-testid="QandA"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            id="searchBar"
            onChange={[Function]}
            placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
          />
          <button
            type="submit"
          >
            Search
          </button>
        </form>
        <div
          className="scroll"
        >
          <div>
            <div
              className="question-list"
            >
              <b>
                Q: 
                Why is this product cheaper here than other sites?
              </b>
              <span
                className="questions-helpful"
              >
                Helpful?
                <span
                  className="clickable"
                  onClick={[Function]}
                >
                   
                  <u>
                    Yes
                  </u>
                   
                </span>
                (
                9
                )
                <span
                  onClick={[Function]}
                >
                   | 
                  <u
                    className="clickable"
                  >
                    Add Answer
                  </u>
                </span>
              </span>
            </div>
            <div>
              <span
                className="A-tag"
              >
                A:
              </span>
              <ul
                className="answer-list"
              >
                <span
                  className="answer-list"
                >
                  seller test answer
                </span>
                <div
                  className="answer-footer"
                >
                  <span>
                    by
                     
                    <b>
                      Seller
                    </b>
                  </span>
                  <span>
                    , 
                    2/16/2022
                  </span>
                  <span>
                     | Helpful?
                  </span>
                  <span
                    className="clickable"
                    onClick={[Function]}
                  >
                     
                    <u>
                      Yes
                    </u>
                     
                  </span>
                  <span>
                    (
                    0
                    ) | 
                  </span>
                  <span
                    onClick={[Function]}
                  >
                    <u
                      className="clickable"
                    >
                      Report
                    </u>
                  </span>
                </div>
                <span
                  className="answer-list"
                >
                  This product is overstocked here!
                </span>
                <div
                  className="answer-footer"
                >
                  <span>
                    by
                     
                    toofast
                  </span>
                  <span>
                    , 
                    9/11/2017
                  </span>
                  <span>
                     | Helpful?
                  </span>
                  <span
                    className="clickable"
                    onClick={[Function]}
                  >
                     
                    <u>
                      Yes
                    </u>
                     
                  </span>
                  <span>
                    (
                    11
                    ) | 
                  </span>
                  <span
                    onClick={[Function]}
                  >
                    <u
                      className="clickable"
                    >
                      Report
                    </u>
                  </span>
                </div>
              </ul>
              <div
                className="load-answers"
                onClick={[Function]}
              >
                <b>
                  See more answers
                </b>
              </div>
            </div>
          </div>
          <div>
            <div
              className="question-list"
            >
              <b>
                Q: 
                Where is this product made?
              </b>
              <span
                className="questions-helpful"
              >
                Helpful?
                <span
                  className="clickable"
                  onClick={[Function]}
                >
                   
                  <u>
                    Yes
                  </u>
                   
                </span>
                (
                9
                )
                <span
                  onClick={[Function]}
                >
                   | 
                  <u
                    className="clickable"
                  >
                    Add Answer
                  </u>
                </span>
              </span>
            </div>
            <div>
              <span
                className="A-tag"
              >
                A:
              </span>
              <ul
                className="answer-list"
              >
                <span
                  className="answer-list"
                >
                  China
                </span>
                <div
                  className="answer-footer"
                >
                  <span>
                    by
                     
                    ceasar
                  </span>
                  <span>
                    , 
                    3/5/2017
                  </span>
                  <span>
                     | Helpful?
                  </span>
                  <span
                    className="clickable"
                    onClick={[Function]}
                  >
                     
                    <u>
                      Yes
                    </u>
                     
                  </span>
                  <span>
                    (
                    4
                    ) | 
                  </span>
                  <span
                    onClick={[Function]}
                  >
                    <u
                      className="clickable"
                    >
                      Report
                    </u>
                  </span>
                </div>
                <span
                  className="answer-list"
                >
                  test post answer to see if answer renders
                </span>
                <div
                  className="answer-footer"
                >
                  <span>
                    by
                     
                    test5
                  </span>
                  <span>
                    , 
                    2/16/2022
                  </span>
                  <span>
                     | Helpful?
                  </span>
                  <span
                    className="clickable"
                    onClick={[Function]}
                  >
                     
                    <u>
                      Yes
                    </u>
                     
                  </span>
                  <span>
                    (
                    0
                    ) | 
                  </span>
                  <span
                    onClick={[Function]}
                  >
                    <u
                      className="clickable"
                    >
                      Report
                    </u>
                  </span>
                </div>
              </ul>
              <div
                className="load-answers"
                onClick={[Function]}
              >
                <b />
              </div>
            </div>
          </div>
          <div>
            <div
              className="question-list"
            >
              <b>
                Q: 
                test post question
              </b>
              <span
                className="questions-helpful"
              >
                Helpful?
                <span
                  className="clickable"
                  onClick={[Function]}
                >
                   
                  <u>
                    Yes
                  </u>
                   
                </span>
                (
                7
                )
                <span
                  onClick={[Function]}
                >
                   | 
                  <u
                    className="clickable"
                  >
                    Add Answer
                  </u>
                </span>
              </span>
            </div>
            <div>
              <br />
              <ul
                className="answer-list"
              />
              <div
                className="load-answers"
                onClick={[Function]}
              >
                <b />
              </div>
            </div>
          </div>
          <div>
            <div
              className="question-list"
            >
              <b>
                Q: 
                another test post question
              </b>
              <span
                className="questions-helpful"
              >
                Helpful?
                <span
                  className="clickable"
                  onClick={[Function]}
                >
                   
                  <u>
                    Yes
                  </u>
                   
                </span>
                (
                5
                )
                <span
                  onClick={[Function]}
                >
                   | 
                  <u
                    className="clickable"
                  >
                    Add Answer
                  </u>
                </span>
              </span>
            </div>
            <div>
              <br />
              <ul
                className="answer-list"
              />
              <div
                className="load-answers"
                onClick={[Function]}
              >
                <b />
              </div>
            </div>
          </div>
          <button
            className="cust-button"
            onClick={[Function]}
          >
            More Answered Questions
          </button>
          <button
            className="cust-button"
            onClick={[Function]}
          >
            Add a Question
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
